docs/content/guides/dagster/branch-depl-with-snowflake.mdx

```diff
@@ -0,0 +1,348 @@
+    result.rename(columns={"by": "user_id"}, inplace=True)
+    return result
+
+
+@asset(
+    io_manager_key="snowflake_io_manager",
+)
+def comments(items: pd.DataFrame) -> pd.DataFrame:
+    """Comments from the Hacker News API."""
+    return items[items["type"] == "comment"]
+
+
+@asset(
+    io_manager_key="snowflake_io_manager",
+)
+def stories(items: pd.DataFrame) -> pd.DataFrame:
+    """Stories from the Hacker News API."""
+    return items[items["type"] == "story"]
+```
+
+As you can see, our assets use a [I/O manager](/concepts/io-management/io-managers) named `snowflake_io_manager` (source code available in the [Transitioning Data Pipelines from Development to Production](/guides/dagster/local-to-production-tutorial#part-one-local-development) guide). Using I/O managers and other resources allow us to swap out implementations per environment without modifying our business logic.
+
+---
+
+## Step 2: Configure our job for each environment
+
+At runtime, we’d like to determine which environment our code is running in: branch deployment, or production. This information dictates how our code should execute, specifically with which credentials and with which database.
+
+To ensure we can't accidentally write to production from within our branch deployment, we’ll use a different set of credentials from production and write to our database clone.
+
+Dagster automatically sets certain environment variables containing deployment metadata, allowing us to read these environment variables to discern between deployments. We can access the `DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT` environment variable to determine the currently executing environment. Refer to [TODO: PAGE TITLE]() for a full list of preset environment variables.
+
+Because we want to configure our assets to write to Snowflake using a different set of credentials and database in each environment, we’ll configure a separate I/O manager for each environment:
+
+```python file=/guides/dagster/local_to_production/branch_depl_tutorial/repository_v1.py startafter=start_repository endbefore=end_repository
+@repository
+def repo():
+    snowflake_config = {
+        "account": {"env": "SNOWFLAKE_ACCOUNT"},
+        "user": {"env": "SNOWFLAKE_USER"},
+        "password": {"env": "SNOWFLAKE_PASSWORD"},
+        "schema": "HACKER_NEWS",
+    }
+    resource_defs = {
+        "branch": {
+            "snowflake_io_manager": snowflake_io_manager.configured(
+                {
+                    **snowflake_config,
+                    "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+                }
+            ),
+        },
+        "production": {
+            "snowflake_io_manager": snowflake_io_manager.configured(
+                {
+                    **snowflake_config,
+                    "database": "PRODUCTION",
+                }
+            ),
+        },
+    }
+
+    def get_current_env():
```

Comment:
should we come up with a function that will return `production` `local` or `branch` ?

Reply:
Hm.. this is what I had originally, but if we added `local`, we'd also have to define unused resources for local development. I think the feedback there was that adding local development was unnecessary and would add bloat to the guide

Reply:
yeah bloating the config would be bad.

I'm just wondering if somewhere we should have a suggested way for people to determine local vs branch vs prod. something like 

```
def get_current_env():
     is_branch_depl = os.getenv("DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT")
     if is_branch_depl:
         return "branch"
     else:
            return os.getenv("DAGSTER_DEPLOYMENT", "local")
```

maybe it wouldn't be written in this guide, but if you're making a full code example of the guide it could be in there with the config for every deployment type and then we can just have a note in the guide saying to look there

Reply:
Yeah this is a good call. After the guide is close to done, I'll add a full installable code example that also includes local development